### DISCUSSION TITLE
The type of 'xProvider' can't be inferred because it depends on itself through the cycle

### DISCUSSION THREAD

**Discussion by shahmirzali49** [rrousselGit/riverpod#3084] [Q&A] [answered]
What is the problem here? can someone explain how can I fix the issue ?

<img width="1057" alt="Screenshot 2023-11-01 at 23 09 10" src="https://github.com/rrousselGit/riverpod/assets/106958390/a1c805dd-cabd-4465-8bc9-ddcab6c51b50">

```
The type of 'netwokServiceProvider' can't be inferred because it depends on itself through the cycle: adressServicesProvider, loginAgainProvider, netwokServiceProvider.
Try adding an explicit type to one or more of the variables in the cycle in order to break the cycle.
```

```
final netwokServiceProvider = Provider<NetworkService>(
  (ref) {
    final Dio dio = Dio();
    return NetworkService(
      dio: dio,
      onUnauthorized: () {
        ref.read(loginAgainProvider.notifier).loginAgain();
      },
    );
  },
);

```

**Comment by rrousselGit** [comment, marked as answer]
You have a circular dependency on your provider.
This causes Dart's type inference to fail.

You can fix the type-inference issue by explicitly typing your providers:


```dart
final Provider<int> provider = Provider<int>(...);
```

Note that circular dependencies are not allowed. Chances are you'd get a runtime exception when fixing that Dart error if you did indeed have a circular dependency

**Reply by shahmirzali49** [reply]
thank you for your answer
